pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```
@@ -346,8 +348,18 @@ private boolean invalidOriginalPrincipal(String originalPrincipal) {
             } else {
                 isProxyAuthorizedFuture = CompletableFuture.completedFuture(true);
             }
-            isAuthorizedFuture = service.getAuthorizationService().allowTopicOperationAsync(
-                topicName, operation, authRole, authenticationData);
+            isAuthorizedFuture = CompletableFuture.supplyAsync(() -> {
+                for (String authRole : authRoles) {
+                    try {
+                        if (service.getAuthorizationService().allowTopicOperationAsync(
+                                topicName, operation, authRole, authenticationData).get()) {
+                            return true;
+                        }
+                    } catch (InterruptedException | ExecutionException e) {
```

Will this hide any errors we could be interested in?